src/history/test/HistoryTestsUtils.cpp

```diff
@@ -615,14 +616,9 @@ CatchupSimulation::catchupOffline(Application::pointer app, uint32_t toLedger)
     auto catchupConfiguration =
         CatchupConfiguration{toLedger, app->getConfig().CATCHUP_RECENT,
                              CatchupConfiguration::Mode::OFFLINE};
-    lm.startCatchup(catchupConfiguration);
-    REQUIRE(!app->getClock().getIOContext().stopped());
-
-    auto caughtUp = [&]() { return lm.isSynced(); };
-    crankUntil(app, caughtUp, std::chrono::seconds{30});
-
-    auto result = caughtUp();
-    if (result)
+    Json::Value val;
+    bool success = catchup(app, catchupConfiguration, val) == 0;
```

Comment:
this change removes the timeout check, which doesn't seem desirable.
I would add a timeout parameter to `catchup` if we really want to do this

Reply:
Ok, brought the timeout back -- the condition for completion just mimics one for offline catchup (it's much faster to just check for catchup failure instead of waiting for 30 seconds timeout)